framework/master-key-interface.md

```diff
@@ -0,0 +1,96 @@
+[//]: # (Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.)
+[//]: # (SPDX-License-Identifier: CC-BY-SA-4.0)
+
+# Master Key Interface
+
+## Version
+
+0.1.0-preview
+
+## Implementations
+
+- [Java](https://github.com/aws/aws-encryption-sdk-java/blob/master/src/main/java/com/amazonaws/encryptionsdk/MasterKey.java)
+- [Python](https://github.com/aws/aws-encryption-sdk-python/blob/master/src/aws_encryption_sdk/key_providers/base.py)
+
+## Overview
+
+Master keys are master key providers that only provide themselves.
+They also support generating and encrypting data keys.
+
+### Legacy
+
+This is a legacy specification.
+Master keys SHOULD NOT be included in any additional implementations.
+Any new implementations MUST include [Keyrings](./keyring-interface.md) instead.
+
+### Consistency
+
+This specification defines the common behavior between the two implementations
+that determine the REQUIRED functionality.
+
+Other specifics of behavior and API vary between the two implementations.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted
+as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Interface
+
+### Get Master Key
+
+Inputs and outputs are the same as for [master key providers](./master-key-provider-interface.md).
+
+A master key MUST supply itself and MUST NOT supply any other master keys.
+
+### Get Master Keys For Encryption
+
+Inputs and outputs are the same as for [master key providers](./master-key-provider-interface.md).
+
+A master key MUST supply itself and MUST NOT supply any other master keys.
+
+### Decrypt Data Key
+
+Inputs and outputs are the same as for [master key providers](./master-key-provider-interface.md).
+
+A master key MUST not attempt to use any other master keys than itself.
```

Comment:
I think this is clearer if we remove "than itself" from the end of the sentence

Reply:
sure